packages/rest/test/unit/coercion/utils.ts

```diff
@@ -78,18 +72,24 @@ export async function testCoercion<T>(config: TestArgs<T>) {
   }
 }
 
-// tslint:disable-next-line:no-any
-export function test(t: any[]) {
+export function test<T>(
+  paramSpec: ParameterObject,
+  rawValue: string | undefined | object,
+  expectedResult: T,
+  opts?: TestOptions,
+) {
   const caller: string = new Error().stack!;
-  it(t[0] as string, async () => {
+  const DEFAULT_TEST_NAME = `convert request raw value ${rawValue} to ${expectedResult}`;
+  const testName = (opts && opts.testName) || DEFAULT_TEST_NAME;
+
+  it(testName, async () => {
     await testCoercion({
-      schema: t[1] as SchemaObject,
-      valueFromReq: t[2] as string,
-      expectedResult: t[3],
+      paramSpec,
+      rawValue,
+      expectedResult,
       caller,
```

Comment:
Do we still need to pass `caller`? I thought you wrote the original stack trace already points to the actual test case?

Reply:
Yep we still need it. function `test` is called directly in a test file, so `new Error().stack` shows the caller.
`test` calls the `testCoercion` and wraps it with `it('test name', () => { testCoercion()})`, inside function `testCoercion` the error stack doesn't know the caller in a test file, so it still relies `test` passes the caller to it.